src/ast/src/lib.rs

```diff
@@ -17,12 +17,60 @@ use std::{
     ops::{Add, BitAnd, BitOr, Div, Mul, Rem, Shl, Shr, Sub},
 };
 
-///! Runtime cell and row representation.
+use crate::values::{Bool, ScalarValue};
 use bigdecimal::ToPrimitive;
 use ordered_float::OrderedFloat;
-use sqlparser::ast::Value;
-
 use sql_model::sql_types::SqlType;
+use sqlparser::ast::{DataType, Expr, Value};
+use std::fmt::{self, Display, Formatter};
+
+pub mod scalar;
+pub mod values;
+
+#[derive(Debug, PartialEq)]
+pub enum ScalarError {
+    NotSupportedType(DataType),
+    NotSupportedValue(Value),
+    NotHandled(Expr),
+}
+
+impl Display for ScalarError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            ScalarError::NotSupportedType(data_type) => write!(f, "not supported type '{}'", data_type),
+            ScalarError::NotSupportedValue(value) => write!(f, "not supported value '{}'", value),
+            ScalarError::NotHandled(expr) => write!(f, "not handled Expression [{}]", expr),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct OperationError(Operation, Option<String>);
+
+impl Display for OperationError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "operation '{}' not supported", self.0)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Operation {
```

Comment:
Uh, I should think on better name because it is for error reporting purpose. It should be used only with `OperationError`, at least this is my intent.